packages-node/providence-analytics/src/program/utils/get-source-code-fragment-of-declaration.js

```diff
@@ -0,0 +1,98 @@
+const fs = require('fs');
+const babelTraversePkg = require('@babel/traverse');
+const { AstService } = require('../services/AstService.js');
+
+/**
+ *  Assume we had:
+ *  ```js
+ *  const x = 88;
+ *  const y = x;
+ *  export const myIdentifier = y;
+ *  ```
+ *  - We started in getSourceCodeFragmentOfDeclaration (looing for 'myIdentifier'), which found VariableDeclarator of export myIdentifier
+ *  - getReferencedDeclaration is called with { referencedIdentifierName: 'y', ... }
+ *  - now we will look in globalScopeBindings, till we find declaration of 'y'
+ *    - Is it a ref? Call ourselves with referencedIdentifierName ('x' in example above)
+ *    - is it a non ref declaration? Return the path of the node
+ * @param {{ referencedIdentifierName:string, globalScopeBindings:BabelBinding; }} opts
+ * @returns {BabelNodePath}
+ */
+function getReferencedDeclaration({ referencedIdentifierName, globalScopeBindings }) {
+  const [, refDeclaratorBinding] = Object.entries(globalScopeBindings).find(
+    ([key]) => key === referencedIdentifierName,
+  );
+
+  if (refDeclaratorBinding.path.node.init.type === 'Identifier') {
```

Comment:
Maybe do below instead:
```js
const referencedIdentifierName = refDeclaratorBinding.path.node.init.name;
if (referencedIdentifierName) {
  return getReferencedDeclaration({ referencedIdentifierName, globalScopeBindings });
}
```
And if it is not guaranteed to deference `refDeclaratorBinding.path.node.init.name`, better to use `?.` to access properties

Reply:
At this point we can trust that `refDeclaratorBinding.path.node.init` exists (it has a name in case it's an Identifier)